optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/employeeroster/EmployeeAvailabilityBlobPopoverContent.java

```diff
@@ -144,9 +149,36 @@ public void onCloseButtonClick(@ForEvent("click") final MouseEvent e) {
     public void onApplyButtonClick(@ForEvent("click") final MouseEvent e) {
 
         final EmployeeBlob blob = (EmployeeBlob) blobView.getBlob();
-        final EmployeeAvailabilityView availability = blob.getEmployeeAvailabilityView();
+        final EmployeeAvailabilityView availabilityView = blob.getEmployeeAvailabilityView();
 
-        final Long oldEmployeeId = availability.getEmployeeId();
+        final EmployeeAvailabilityState oldState = availabilityView.getState();
+        final LocalDateTime oldStartDateTime = availabilityView.getStartDateTime();
+        final LocalDateTime oldEndDateTime = availabilityView.getEndDateTime();
+        
+        try {
+            availabilityView.setState(EmployeeAvailabilityState.values()[availabilitySelect.getSelectedIndex()]);
+            availabilityView.setStartDateTime(from.getValue());
+            availabilityView.setEndDateTime(to.getValue());
+        } catch (ValidationException invalidField) {
```

Comment:
@mbarkley is this normal flow control in Errai?

Reply:
Typically, this is done using a `DataBinder`, which will set the fields for you (provided you give the widget <-> field mapping).

Reply:
Yeah, the Errai way is to use data-binding and Errai validation, but neither of those are used to make UI in the workbench either so this approach looks fine to me.